src/extension/mapml/src/test/java/org/geoserver/mapml/MapMLWMSFeatureTest.java

```diff
@@ -60,38 +88,140 @@ public void tearDown() {
+        Catalog cat = getCatalog();
+        LayerInfo li = cat.getLayerByName(MockData.BUILDINGS.getLocalPart());
+        li.getMetadata().put(MAPML_USE_FEATURES, true);
+        li.getMetadata().put(MAPML_USE_TILES, false);
+        li.getStyles().add(cat.getStyleByName("polygonFilter"));
+        li.getStyles().add(cat.getStyleByName("polygonElseFilter"));
+        li.setDefaultStyle(cat.getStyleByName("polygonFilter"));
+        cat.save(li);
+        Mapml mapmlFeatures =
+                getWMSAsMapML(
+                        MockData.BUILDINGS.getLocalPart(),
+                        null,
+                        null,
+                        null,
+                        "EPSG:4326",
+                        "polygonFilter",
+                        true);
+
+        assertEquals(
+                "Buildings layer has two features, only one should show up after the SLD is applied",
+                1,
+                mapmlFeatures.getBody().getFeatures().size());
+
+        Mapml mapmlFeaturesElse =
+                getWMSAsMapML(
+                        MockData.BUILDINGS.getLocalPart(),
+                        null,
+                        null,
+                        null,
+                        "EPSG:4326",
+                        "polygonElseFilter",
+                        true);
+
+        assertEquals(
+                "Buildings layer has two features, both should show up after the SLD with elseFilter is applied",
+                2,
+                mapmlFeaturesElse.getBody().getFeatures().size());
+    }
+
+    @Test
+    public void testMapMLGetStyleQuery() throws Exception {
+        Catalog cat = getCatalog();
+        MemoryDataStore ds = new MemoryDataStore();
+        final String polyTypeSpec = "ADDRESS:String,ip:Integer,geom:Polygon:srid=4326";
+        SimpleFeatureType polyType = DataUtilities.createType("polygons", polyTypeSpec);
+        ds.addFeature(
+                feature(
+                        polyType,
+                        "polygon1",
+                        "123 Main Street",
+                        1000,
+                        "POLYGON ((1 1, 2 2, 3 3, 4 4, 1 1))"));
+        ds.addFeature(
+                feature(
+                        polyType,
+                        "polygon2",
+                        "95 Penny Lane",
+                        2000,
+                        "POLYGON ((6 6, 7 7, 8 8, 9 9, 6 6))"));
+        ds.addFeature(
+                feature(
+                        polyType,
+                        "polygon3",
+                        "154 Sesame Street",
+                        3000,
+                        "POLYGON ((11 11, 12 12, 13 13, 14 14, 11 11))"));
+        ReferencedEnvelope mapBounds =
+                new ReferencedEnvelope(
+                        0, 0.005, 0, 0.005, CRS.decode("urn:x-ogc:def:crs:EPSG:4326"));
+        Rectangle renderingArea = new Rectangle(256, 256);
```

Comment:
Given you're testing the query filter only, do you actually need to have features in the sample data source? 
A ``DataUtilities.store(new EmptyFeatureColection(targetType))``might work as well.

Reply:
changed